File: app/src/main/java/nl/rsdt/japp/jotial/data/structures/area348/AutoInzittendeInfo.java
package nl.rsdt.japp.jotial.data.structures.area348;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Created by mattijn on 30/09/17.
 */

public class AutoInzittendeInfo implements Parcelable{

    public int id;
    public String datetime;
    public String gebruikersNaam;
    public int gebruikersID;
    public String autoEigenaar;
    public String rol;
    /**
     * Initializes a new instance of BaseInfo from the parcel.
     *
     * @param in
     */
    protected AutoInzittendeInfo(Parcel in) {
        id = in.readInt();
        datetime = in.readString();
        gebruikersNaam = in.readString();
        gebruikersID = in.readInt();
        autoEigenaar = in.readString();
        rol = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int fags) {
        dest.writeInt(id);
        dest.writeString(datetime);
        dest.writeString(gebruikersNaam);
        dest.writeInt(gebruikersID);
        dest.writeString(autoEigenaar);
        dest.writeString(rol);
    }
    public static final Creator<AutoInzittendeInfo> CREATOR = new Creator<AutoInzittendeInfo>() {
        @Override
        public AutoInzittendeInfo createFromParcel(Parcel in) {
            return new AutoInzittendeInfo(in);
        }

        @Override
        public AutoInzittendeInfo[] newArray(int size) {
            return new AutoInzittendeInfo[size];
        }
    };
}

File: app/src/main/java/nl/rsdt/japp/jotial/data/structures/area348/AutoInzittendeInfo.kt
package nl.rsdt.japp.jotial.data.structures.area348

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by mattijn on 30/09/17.
 */

class AutoInzittendeInfo
/**
 * Initializes a new instance of BaseInfo from the parcel.
 *
 * @param in
 */
protected constructor(`in`: Parcel) : Parcelable {

    var id: Int = 0
    var datetime: String? = null
    var gebruikersNaam: String? = null
    var gebruikersID: Int = 0
    var autoEigenaar: String? = null
    var rol: String? = null

    init {
        id = `in`.readInt()
        datetime = `in`.readString()
        gebruikersNaam = `in`.readString()
        gebruikersID = `in`.readInt()
        autoEigenaar = `in`.readString()
        rol = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, fags: Int) {
        dest.writeInt(id)
        dest.writeString(datetime)
        dest.writeString(gebruikersNaam)
        dest.writeInt(gebruikersID)
        dest.writeString(autoEigenaar)
        dest.writeString(rol)
    }

    companion object {
        val CREATOR: Parcelable.Creator<AutoInzittendeInfo> = object : Parcelable.Creator<AutoInzittendeInfo> {
            override fun createFromParcel(`in`: Parcel): AutoInzittendeInfo {
                return AutoInzittendeInfo(`in`)
            }

            override fun newArray(size: Int): Array<AutoInzittendeInfo> {
                return arrayOfNulls(size)
            }
        }
    }
}
